src/datasets/datasets.service.ts

```diff
@@ -88,16 +100,33 @@ export class DatasetsService {
         "pid",
         filter.fields as FilterQuery<DatasetDocument>,
       );
+
     const whereClause: FilterQuery<DatasetDocument> = {
       ...filterQuery,
       ...extraWhereClause,
     };
     const modifiers: QueryOptions = parseLimitFilters(filter.limits);
+    const { $text, ...remainingClauses } = whereClause;
+    if (!this.ESClient || !$text) {
+      const datasets = await this.datasetModel
+        .find(whereClause, null, modifiers)
+        .exec();
+      return datasets;
+    }
+
+    const esResult = await this.ESClient.search(
+      filter.fields as IDatasetFields,
+      modifiers.limit,
+      modifiers.skip,
+    );
 
     const datasets = await this.datasetModel
-      .find(whereClause, null, modifiers)
+      .find(
+        { _id: { $in: esResult.data }, ...remainingClauses },
+        null,
+        modifiers,
+      )
```

Comment:
It seems to me that here the elastic search is always used in the search.
What happens if we disable elastic search in configuration?

Reply:
```
    if (!this.ESClient || !$text) {
      const datasets = await this.datasetModel
        .find(whereClause, null, modifiers)
        .exec();
      return datasets;
    }
```
if we disable elastic search `!this.ESClient` will be true and it will execute only within the block